nexus/src/app/background/blueprint_load.rs

```diff
@@ -0,0 +1,176 @@
+                    warn!(
+                        &log,
+                        "failed to read target blueprint";
+                        "error" => &message
+                    );
+                    let e =
+                        format!("failed to read target blueprint: {message}");
+                    json!({"error": e})
+                }
+                (None, Ok(None)) => {
+                    // We haven't found a blueprint yet. Do nothing.
+                    json!({"status": "no target blueprint"})
+                }
+                (Some(old), Ok(None)) => {
+                    // We have transitioned from having a blueprint to not
+                    // having one. This should not happen.
+                    let message = format!(
+                        "target blueprint with id {} was removed. There is no \
+                        longer any target blueprint",
+                        old.id
+                    );
+                    let old_id = old.id.to_string();
+                    self.last = None;
+                    self.tx.send_replace(self.last.clone());
+                    error!(&log, "{message:?}");
+                    json!({
+                        "removed_target_id": old_id,
+                        "status": "no target blueprint (removed)",
+                        "error": message
+                    })
+                }
+                (None, Ok(Some((_, new_target)))) => {
+                    // We've found a target blueprint for the first time.
+                    // Save it and notify any watchers.
+                    let target_id = new_target.id.to_string();
+                    let time_created = new_target.time_created.to_string();
+                    info!(
+                        log,
+                        "found new target blueprint (first find)";
+                        "target_id" => &target_id,
+                        "time_created" => &time_created
+                    );
+                    self.last = Some(Arc::new(new_target));
+                    self.tx.send_replace(self.last.clone());
+                    json!({
+                        "target_id": target_id,
+                        "time_created": time_created,
+                        "time_found": chrono::Utc::now().to_string(),
+                        "status": "first target blueprint"
+                    })
+                }
+                (Some(old), Ok(Some((_, new)))) => {
+                    let target_id = new.id.to_string();
+                    let time_created = new.time_created.to_string();
+                    if old.id != new.id {
+                        // The current target blueprint has been updated
+                        info!(
+                            log,
+                            "found new target blueprint";
+                            "target_id" => &target_id,
+                            "time_created" => &time_created
```

Comment:
Tiny nit - I'd maybe prefix these as `new_target_id` and `new_time_created` to contrast with the `current_*` properties set on the logger when we created it above.

Reply:
I dislike prefixing all the properties. What if I changed `current_*` to `original_*` and left everything else as is.